syntax = "proto3";
package admin;

service AdminService {
  rpc CreateIndex(CreateIndexRequest) returns(CreateIndexResponse) {}
}

message CreateIndexRequest { IndexSchema index_schema = 1; }
message IndexSchema {
  string index_name = 1;
  repeated FieldConfig field_configs = 2;
}
message FieldConfig {
  string field_name = 1;
  FieldType field_type = 2;
}
enum FieldType {
  FIELD_TYPE_INVALID = 0; FIELD_TYPE_STRING_ANALYZED = 1;
  FIELD_TYPE_STRING_NON_ANALYZED = 2;
  FIELD_TYPE_INT = 3;
  FIELD_TYPE_DOUBLE = 4;
  FIELD_TYPE_VECTOR = 5;
}

message CreateIndexResponse { string response_code = 1; }
